File: 0058-length-of-last-word/0058-length-of-last-word.cs
int lengthOfLastWord(char * s){
    int length = strlen(s);
    int count = 0;
    int pos = length - 1;

    // Skip trailing spaces if any
    while (pos >= 0 && s[pos] == ' ') {
        --pos;
    }

    // Count characters in the last word
    while (pos >= 0 && s[pos] != ' ') {
        --pos;
        ++count;
    }

    return count;
}

File: 0058-length-of-last-word/0058-length-of-last-word.cpp
#include<string>

class Solution {
public:
    int lengthOfLastWord(std::string s) {
        int length = 0;
        int pos = s.length() - 1;

        // Skip trailing spaces if any
        while (pos >= 0 && s[pos] == ' ') {
            --pos;
        }

        // Count characters in the last word
        while (pos >= 0 && s[pos] != ' ') {
            --pos;
            ++length;
        }

        return length;
    }
};
